src/emu/machine.cpp

```diff
@@ -424,6 +410,31 @@ int running_machine::run(bool quiet)
 	return error;
 }
 
+//-------------------------------------------------
+//  run_frame - execute one frame for this machine
+//-------------------------------------------------
+
+int running_machine::run_frame(attotime frametime)
+{
+	g_profiler.start(PROFILER_EXTRA);
+
+	// execute CPUs if not paused
+	if (!m_paused) {
+		attotime stoptime(m_scheduler.time() + frametime);
+		while (m_scheduler.time() < stoptime) {
```

Comment:
This loop will never be entered in the non-Emscripten case because `m_scheduler.time() < stoptime` will never be true when `stoptime` is `m_scheduler.time() + attotime::zero`.  Also, if you want the loop to be entered at least once, should use `do ... while ()`.